packages/foam-vscode/src/dated-notes.ts

```diff
@@ -17,13 +18,24 @@ async function openDailyNoteFor(date?: Date) {
   );
   await focusNote(dailyNotePath, isNew);
 }
-function getDailyNotePath(configuration: WorkspaceConfiguration, date: Date) {
-  const rootDirectory = workspace.workspaceFolders[0].uri.fsPath;
-  const dailyNoteDirectory: string =
-    configuration.get('openDailyNote.directory') ?? '.';
+
+function getDailyNotePath(
+  configuration: WorkspaceConfiguration,
+  date: Date
+): URI {
+  const dailyNoteDirectory: URI =
+    URI.file(configuration.get('openDailyNote.directory')) ?? URI.file('.');
   const dailyNoteFilename = getDailyNoteFileName(configuration, date);
 
-  return join(rootDirectory, dailyNoteDirectory, dailyNoteFilename);
+  const rootDirectory = isAbsolute(dailyNoteDirectory.path)
+    ? URI.file('.')
+    : workspace.workspaceFolders[0].uri; // TODO: This isn't guaranteed to exist (They could have no workspace open)
```

Comment:
remove TODO and handle the case.

also, shouldn't this ternary operator be the other way around?

What does `URI.file('.')` resolves to?

Reply:
> remove TODO and handle the case.

I was mistaken. I was misreading a different error message at the time. This seems to be fine (indeed, I didn't change this code from what was there).

> shouldn't this ternary operator be the other way around?

No. I'm was trying to get a URI I can prepend as the first element of `URI.joinPath`. In the case of an absolute path, I needed a URI that would have no effect, so I forget how, but I found that `URI.file('.')` can serve that purpose.

I agree that it could be confusing, so I've rewritten this as a forking return statement.